server/src/main/java/org/opensearch/search/aggregations/bucket/terms/GlobalOrdinalsStringTermsAggregator.java

```diff
@@ -136,16 +144,100 @@ public GlobalOrdinalsStringTermsAggregator(
                 return new DenseGlobalOrds();
             });
         }
+        this.fieldName = (valuesSource instanceof ValuesSource.Bytes.WithOrdinals.FieldData)
+            ? ((ValuesSource.Bytes.WithOrdinals.FieldData) valuesSource).getIndexFieldName()
+            : null;
     }
 
     String descriptCollectionStrategy() {
         return collectionStrategy.describe();
     }
 
+    public void setWeight(Weight weight) {
+        this.weight = weight;
+    }
+
+    /**
+     Collects term frequencies for a given field from a LeafReaderContext directly from stored segment terms
+     @param ctx The LeafReaderContext to collect terms from
+     @param globalOrds The SortedSetDocValues for the field's ordinals
+     @param ordCountConsumer A consumer to accept collected term frequencies
+     @return A no-operation LeafBucketCollector implementation, since collection is complete
+     @throws IOException If an I/O error occurs during reading
+     */
+    LeafBucketCollector termDocFreqCollector(
+        LeafReaderContext ctx,
+        SortedSetDocValues globalOrds,
+        BiConsumer<Long, Integer> ordCountConsumer
+    ) throws IOException {
+        if (weight == null) {
+            // Weight not assigned - cannot use this optimization
+            return null;
```

Comment:
Does this ever get hit? 

I see it's not covered by tests. If you remove this condition, does anything break?

Reply:
I guess nothing breaks, I added this initially as a safeguard, but we don't need this. Given we have UTs as well to cover the scenarios in case this gets null, which it doesn't, as the default implementation of -1 does kicks in. I'll remove this check in the revision.